stix_shifter_modules/vectra/stix_transmission/connector.py

```diff
@@ -0,0 +1,215 @@
+from stix_shifter_utils.modules.base.stix_transmission.base_json_sync_connector import BaseJsonSyncConnector
+from stix_shifter_utils.utils.error_response import ErrorResponder
+from stix_shifter_utils.utils import logger
+from .api_client import APIClient
+import json
+
+
+class Connector(BaseJsonSyncConnector):
+    VECTRA_MAX_PAGE_SIZE = 5000
+
+    def __init__(self, connection, configuration):
+        self.api_client = APIClient(connection, configuration)
+        self.logger = logger.set_logger(__name__)
+        self.connector = __name__.split('.')[1]
+
+    async def ping_connection(self):
+        return_obj = {}
+        response_dict = {}
+        try:
+            response_wrapper = await self.api_client.ping_data_source()
+            response_code = response_wrapper.code
+            if response_code == 200:
+                return_obj['success'] = True
+            elif response_code in (401, 403):
+                response_dict['code'] = response_code
+                response_dict['message'] = "Invalid Authentication"
+                ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+        except Exception as ex:
+            if "timeout_error" in str(ex):
+                response_dict['code'] = 408
+            response_dict['message'] = str(ex)
+            self.logger.error('error when getting search results: %s', str(ex))
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+        return return_obj
+
+    async def create_results_connection(self, query, offset, length):
+        """
+         Function to create query connection
+        :param query: str
+        :param offset: int
+        :param length: str
+        :return: list
+        """
+        response_dict = {}
+        return_obj = {}
+        data = []
+        try:
+            offset = int(offset)
+            length = int(length)
+            pagination_offset = offset
+            total_records = length
+            page = 1
+
+            # changing offset to page number
```

Comment:
can you explain this calculation? what is `pagination_offset`? why does it needs to be less than 5? what is the difference between page and page_size?

Reply:
Previous page calculation corrected, please check the below example 

Vectra doesn't support offset values in queries, we have the option to mention page(page number) and page size.
We need to find out given offset is under which page and then we will fetch the particular page.
example: inputs offset=2010 and length=2000
		converting offset 2010 to page = 2, page_size = 2000 and pagination_offset = 10				
		Page and page_size(length) are API parameters for the query.
		pagination_offset is need for slicing.
		
Here, offset 2010 is under page 2. So, will fetch the page 2, which contains 2000 records.
From the records, we are slicing from 10(pagination_offset) to 2000.

Reply:
fair calculation. but just wondering, does `next_url` contains page,page_size for next set of results? I wonder if you can make use of next_url to retrieve next batch of results in the subsequent calls.

Reply:
yes, next_url contains query, page, and page_size.

Reply:
I wonder why you are not using next_url for pagination. the calculation generates all the pagination parameters that are already contains in next_url. The calculation is needed for the first call but for subsequent calls until the result limit you can use next_url. let me know if i am missing something!

Reply:
In next_url, we are not doing any calculations, only splitting the next_url because it contains the host name.
When making an API call, we should only pass the API endpoint with parameters. host name value set at the init function.

Reply:
as discussed `metadata` parameter needs to be used in order to save the pagination url and results count.

Reply:
Pagination handled using metadata